test/unit-tests/components/views/settings/encryption/ResetIdentityPanel-test.tsx

```diff
@@ -30,7 +31,22 @@ describe("<ResetIdentityPanel />", () => {
         );
         expect(asFragment()).toMatchSnapshot();
 
-        await user.click(screen.getByRole("button", { name: "Continue" }));
+        // We need to pause the reset so that we can check that it's providing
+        // feedback to the user that something is happening.
+        let resolveResetEncryption;
+        const resetEncryptionPromise: Promise<void> = new Promise((resolve) => {
+            resolveResetEncryption = resolve;
+        });
```

Comment:
We can use `defer` here instead

Reply:
Or https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers

Reply:
Which is built-in, better to use t3chguy option

Reply:
When I try to use `withResolvers`, TypeScript complains, saying that I need to set "the 'lib' compiler option to 'es2024' or later", which I'm not sure if it is something I should do.  So I'll use `defer` for now.